sfdc-accounts-to-dust-new.ts

```
@@ -0,0 +1,422 @@
+  };
+
+  if (SF_USERNAME && SF_PASSWORD && SF_SECURITY_TOKEN) {
+    conn = new jsforce.Connection(connectionOptions);
+    await conn.login(SF_USERNAME, SF_PASSWORD + SF_SECURITY_TOKEN);
+  } else if (SF_CLIENT_ID && SF_CLIENT_SECRET) {
+    conn = new jsforce.Connection({
+      ...connectionOptions,
+      oauth2: {
+        clientId: SF_CLIENT_ID,
+        clientSecret: SF_CLIENT_SECRET,
+        loginUrl: SF_LOGIN_URL,
+      },
+    });
+    await (conn.oauth2.authenticate as any)({
+      grant_type: "client_credentials",
+      client_id: SF_CLIENT_ID,
+      client_secret: SF_CLIENT_SECRET,
+    });
+  } else {
+    throw new Error("Insufficient authentication information provided");
+  }
+  console.log("Connected to Salesforce");
+  console.log("Date of last update:", UPDATED_SINCE);
+  return conn;
+}
+
+async function getRecentlyUpdatedAccountIds(
+  conn: jsforce.Connection
+): Promise<Set<string>> {
+  const queries = [
+    `SELECT Id FROM Account WHERE LastModifiedDate >= ${UPDATED_SINCE}`,
+    `SELECT AccountId FROM Contact WHERE LastModifiedDate >= ${UPDATED_SINCE}`,
+    `SELECT AccountId FROM Opportunity WHERE LastModifiedDate >= ${UPDATED_SINCE}`,
+    `SELECT AccountId FROM Case WHERE LastModifiedDate >= ${UPDATED_SINCE}`,
+  ];
+  const accountIds = new Set<string>();
+  for (const query of queries) {
+    console.log(`Executing query: ${query}`);
+    try {
+      let result = await conn.query<{ Id: string; AccountId?: string }>(query);
+      console.log(`Found ${result.totalSize} records`);
+      do {
+        result.records.forEach((record) => {
+          accountIds.add(record.Id || record.AccountId || "");
+        });
+        if (!result.done) {
+          result = await conn.queryMore<{ Id: string; AccountId?: string }>(
+            result.nextRecordsUrl
+          );
+        }
+      } while (!result.done);
+    } catch (error) {
+      console.error(`Error executing query: ${query}`, error);
+    }
```

[EH-003] Do not suppress errors in catch blocks. Consider using Promise.allSettled for handling multiple queries and create a custom SalesforceQueryError class to properly handle and propagate errors.